includes/admin/tasks/class-wc-payments-task-disputes.php

```diff
@@ -34,13 +34,38 @@ class WC_Payments_Task_Disputes extends Task {
 	 */
 	private $database_cache;
 
+
+	/**
+	 * Disputes due within 7 days.
+	 *
+	 * @var array|null
+	 */
+	private $disputes_due_within_7d;
+
+	/**
+	 * Disputes due within 1 day.
+	 *
+	 * @var array|null
+	 */
+	private $disputes_due_within_1d;
+
 	/**
 	 * WC_Payments_Task_Disputes constructor.
 	 */
 	public function __construct() {
-		$this->api_client     = \WC_Payments::create_api_client();
+
+		$this->api_client     = \WC_Payments::get_payments_api_client();
 		$this->database_cache = \WC_Payments::get_database_cache();
 		parent::__construct();
+		$this->init();
```

Comment:
Running this `init()` function here will call the task on every page view, so probably worse than before.
Maybe something like
```php
	private $disputes_due_within_7d = null;
	private $disputes_due_within_1d = null;
	private function init() {
        if ( $this->disputes_due_within_7d !== null ) {
            return;
        } 
		$this->disputes_due_within_7d = $this->get_disputes_needing_response_within_days( 7 );
		$this->disputes_due_within_1d = $this->get_disputes_needing_response_within_days( 1 );
	}

    public function get_title() {
        $this->init();
        ...
    }
    public function can_view() {
        $this->init();
        ...
    }
    public function get_additional_info() {
        $this->init();
        ...
    }
```

Looking at the code, `can_view()` is always run before the other methods so it could just be ran here. But might not be obvious in reading.

Reply:
Good point, thanks for the heads up.

Unfortunately, it looks like `can_view()` runs on each (WC) page view as well. 🤔

Reply:
Makes sense, since the class is being instantiated on the `init` hook.

https://github.com/Automattic/woocommerce-payments/blob/b10377c46edc62faed6d9ffaf5716a23dd2db5ce/includes/class-wc-payments-tasks.php#L26

Perhaps a `maybe_add_task_disputes_need_response` action is required here. Or more suitable hook to call.

Reply:
> Unfortunately, it looks like can_view() runs on each (WC) page view as well. 🤔

Doh! I've had a look into this and I cannot see a straightforward solution here. The performance changes are negligible in my opinion compared to the original implementation a year ago.